tools/stats/upload_stats_lib.py

```diff
@@ -213,3 +220,130 @@ def is_rerun_disabled_tests(root: ET.ElementTree) -> bool:
 
     message = skipped.attrib.get("message", "")
     return TARGET_WORKFLOW in message or "num_red" in message
+
+
+def _convert_float_values_to_decimals(data: Dict[str, Any]) -> Dict[str, Any]:
+    for key, value in data.items():
+        if isinstance(value, float):
+            data[key] = decimal.Decimal(str(value))  # str() preserves the precision
+    return data
```

Comment:
Nit (also existing algorithm modifies input in place, is this intentional)
```suggestion
    return {k: Decimal(str(v)) if isinstance(v, float) else v for (k,v) in data.items()}
```